src/utils/MainApi.js

```diff
@@ -0,0 +1,82 @@
+const baseUrl =
+  'https://api.news-explorer-hartsek.students.nomoredomainssbs.ru';
+const headers = {
+  'content-type': 'application/json',
+};
+
+const checkResponse = (res) => {
+  return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+};
+
+export const signup = (email, password, name) => {
+  return fetch(`${baseUrl}/signup`, {
+    method: 'POST',
+    headers: headers,
+    body: JSON.stringify({
+      name: name,
+      email: email,
+      password: password,
+    }),
+  }).then(checkResponse);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request`. And you don’t need to change anything else.